src/Nethermind/Nethermind.Core/BlockToProduce.cs

```diff
@@ -49,5 +49,10 @@ public BlockToProduce(BlockHeader blockHeader, IEnumerable<Transaction> transact
         public BlockToProduce(BlockHeader blockHeader) : base(blockHeader)
         {
         }
+
+        public override Block CreateBlockForProcessing(BlockHeader header) => 
```

Comment:
keep in an external class, remove this responsibility from blocks, do not refer to block production in Nethermind.Core

Reply:
We are using polimorhphism here its simpler than creating factories, but if you insist we can go there